backends/bmv2/simple_switch/simpleSwitch.cpp

```diff
@@ -259,83 +282,105 @@ Util::IJson* ExternConverter_digest::convertExternFunction(
 }
 
 Util::IJson* ExternConverter_resubmit::convertExternFunction(
-    UNUSED ConversionContext* ctxt, UNUSED const P4::ExternFunction* ef,
-    UNUSED const IR::MethodCallExpression* mc, UNUSED const IR::StatOrDecl* s,
+    ConversionContext* ctxt, UNUSED const P4::ExternFunction* ef,
+    const IR::MethodCallExpression* mc, UNUSED const IR::StatOrDecl* s,
     UNUSED const bool emitExterns) {
-    if (mc->arguments->size() != 1) {
-        modelError("Expected 1 argument for %1%", mc);
-        return nullptr;
-    }
-    auto primitive = mkPrimitive("resubmit");
-    auto parameters = mkParameters(primitive);
-    primitive->emplace_non_null("source_info", mc->sourceInfoJsonObj());
-    cstring listName = "resubmit";
-    // If we are supplied a type argument that is a named type use
-    // that for the list name.
-    if (mc->typeArguments->size() == 1) {
-        auto typeArg = mc->typeArguments->at(0);
-        if (typeArg->is<IR::Type_Name>()) {
-            auto origType = ctxt->refMap->getDeclaration(
-                typeArg->to<IR::Type_Name>()->path, true);
-            if (!origType->is<IR::Type_Struct>()) {
-                modelError("%1%: expected a struct type", origType->getNode());
-                return nullptr;
+    if (mc->arguments->size() == 1) {
```

Comment:
this case handles the deprecated method, so it should probably be replaced with an error soon.
The same holds true for the following extern converters.